pkg/process/runner/submitter.go

```diff
@@ -374,15 +380,18 @@ func (s *checkSubmitter) consumePayloads(results *api.WeightedQueue, fwd forward
 			}
 
 			if statuses := readResponseStatuses(result.name, responses); len(statuses) > 0 {
-				if updateRTStatus && s.updateRTStatusCallback != nil {
-					s.updateRTStatusCallback(statuses)
+				if updateRTStatus {
+					select {
+					case s.rtNotifierChan <- statuses:
+					default: // Never block on the rtNotifierChan in case the runner has somehow stopped
```

Comment:
Not sure how easy it would be to add this, but it might be nice to have a unit test for this (i.e. never blocking on the channel send). Seems like we might be able to add a focused test by refactoring the `select` block into its own function and testing just that function, but if it ends up being complicated to do, we can just ignore this and move on